### PR TITLE
Fix/ieee xml

### PR DESCRIPTION
**Pull request by opoudjis**
### Metanorma PR checklist

 - [ ] Breaking changes (list related PRs)
 - [ ] Documentation update required ([create task for this](https://github.com/metanorma/metanorma.org/issues/new))
 - [ ] External dependency introduced ([documentation update need](https://github.com/metanorma/metanorma.org/issues/new))
 - [ ] Gem with native library introduced

### CODE REVIEW

**Comment by opoudjis** [comment]
This is waiting on https://github.com/metanorma/mnconvert/issues/195

I need for you to get this working with validation_against: standards.dtd, where standards.dtd is in the folder

"https://github.com/metanorma/ieee-stl/blob/main/standards-1-7-DTD/standards-1-7-dtd/standards.dtd"

**Comment by CAMOBAP** [comment]
For this particular case as far as I understand DTD comes from the private repo. So it doesn't make sense to implement URL validation on mnconvert.jar side

I think we can add CI-only tests (because CI token can download private repos) so we can do DTD validation tests on CI only

@ronaldtse how do you think will it be worth it?

**Comment by opoudjis** [comment]
It's not *mandatory* to test IEEE XML validation, but I'd like for it to happen. Need this resolved for https://github.com/metanorma/metanorma-ieee/issues/132

**Comment by ronaldtse** [comment]
> For this particular case as far as I understand DTD comes from the private repo. So it doesn't make sense to implement URL validation on mnconvert.jar side
> 
> I think we can add CI-only tests (because CI token can download private repos) so we can do DTD validation tests on CI only
> 
> @ronaldtse how do you think will it be worth it?

@CAMOBAP I agree with this approach of using the private DTD in the GHA jobs.

I have asked IEEE SA whether we can share the DTD publicly, but until they answer positively, we have to use this private GHA approach. Thanks!

**Comment by ronaldtse** [comment]
IEEE SA has answered that they want to keep the DTD private. So we will have to action the GHA approach. @CAMOBAP @Intelligent2013 can you please help facilitate? Thanks!

**Comment by CAMOBAP** [comment]
@ronaldtse I will work on it

**Comment by CAMOBAP** [comment]
Now DTD validation test ready, but the test fails because of validation issues

@opoudjis @Intelligent2013 what are the next steps?

**Comment by Intelligent2013** [comment]
```
.../rice-en.cd.isosts.xml is NOT valid reason:
...
.../rice-en.cd.ieee.xml is NOT valid reason:
```
But in https://github.com/metanorma/mnconvert/actions/runs/3084996936/jobs/4987787985
all checks ended ok....
Let me try to understand this...

**Comment by Intelligent2013** [comment]
The most issues will be fixed in mnconvert v1.24, except one `.../rice-en.cd.ieee.xml is NOT valid reason:`, because it generated from ISO MN XML, not IEEE MN XML. And I didn't test mnconvert for IEEE XML generation from arbitrary MN non-IEEE XML.
I'll try to update mnconvert.

**Comment by Intelligent2013** [comment]
@CAMOBAP validation issues fixed.

**Review by CAMOBAP — APPROVED** [review]

**Comment by ronaldtse** [comment]
@Intelligent2013 but we're not supposed to generate IEEE XML from other flavors. Only MN IEEE docs can output IEEE XML. There is clearly some misunderstanding here somewhere?

**Comment by ronaldtse** [comment]
Anyway the validation passes now, and let's merge it. 

But still we should not allow generating IEEE XML from other flavors. It would be a nightmare to support that.

File: spec/mnconvert_spec.rb [hunk 1]
```diff
@@ -64,6 +64,24 @@
     expect(sts_path.read).to include '<preformat preformat-type="ruby">'
   end
 
+  it "converts XML to IEEE XML", :ci do
+    sts_path = Pathname.new(Dir.pwd) / "rice-en.cd.ieee.xml"
```
Comment (ronaldtse) [inline spec/mnconvert_spec.rb:68]: This test is clearly wrong. We should not convert an Metanorma ISO XML into IEEE XML. We must remove this.

**Pull request merged**